src/zimdump.cpp

```diff
@@ -379,7 +375,7 @@ inline void write_to_file(const std::string &base, const std::string& path, cons
 }
 
 
-void ZimDumper::dumpFiles(const std::string& directory, bool symlinkdump)
+int ZimDumper::dumpFiles(const std::string& directory, bool symlinkdump, std::function<bool (const char c)> nsfilter)
```

Comment:
Do not make the method return a `int` if it always return 0.
It is useless and error prone (caller will assume you return something different than 0 if something goes wrong)

Reply:
I agree. I left `subcmdDumpAll` to return an int thou because it is part of the signature.